plugin-server/tests/worker/ingestion/cookielessServerHashStep.test.ts

```diff
@@ -0,0 +1,371 @@
+import type { PluginEvent } from '@posthog/plugin-scaffold'
+
+import { CookielessServerHashMode, Hub } from '../../../src/types'
+import { closeHub, createHub } from '../../../src/utils/db/hub'
+import { PostgresUse } from '../../../src/utils/db/postgres'
+import { deepFreeze, UUID7 } from '../../../src/utils/utils'
+import {
+    bufferToSessionState,
+    COOKIELESS_MODE_FLAG_PROPERTY,
+    COOKIELESS_SENTINEL_VALUE,
+    cookielessServerHashStep,
+    sessionStateToBuffer,
+    toYYYYMMDDInTimezoneSafe,
+} from '../../../src/worker/ingestion/event-pipeline/cookielessServerHashStep'
+import { createOrganization, createTeam } from '../../helpers/sql'
+
+describe('cookielessServerHashStep', () => {
+    describe('sessionStateToBuffer', () => {
+        it('should return a binary representation of the session state which can be converted back to the original', () => {
+            const date = new Date('2024-12-17T10:50Z')
+            const uuidRand = Buffer.from('0123456789ABCDEF0123', 'hex')
+            const sessionId = new UUID7(date.getTime(), uuidRand)
+
+            // check that the bytes are as expected
+            const sessionStateBuf = sessionStateToBuffer({ sessionId, lastActivityTimestamp: date.getTime() })
+            expect(sessionStateBuf.toString('hex')).toMatchInlineSnapshot(
+                `"0193d43d2fc07123856789abcdef0123c02f3dd493010000"`
+            )
+
+            // make sure its reversible
+            const sessionState = bufferToSessionState(sessionStateBuf)
+            expect(sessionState.lastActivityTimestamp).toEqual(date.getTime())
+            expect(sessionState.sessionId).toEqual(sessionId)
+        })
+    })
+
+    describe('toYYYYMMDDInTimezoneSafe', () => {
+        it('should return a string in the format YYYY-MM-DD giving the date in the given time zone', () => {
+            const date = new Date('2024-12-31T10:00:00Z').getTime()
+            const result = toYYYYMMDDInTimezoneSafe(date, 'Europe/London', 'UTC')
+            expect(result).toEqual('2024-12-31')
+        })
+        it('should handle single digit months and days', () => {
+            const date = new Date('2025-01-01T10:00:00Z').getTime()
+            const result = toYYYYMMDDInTimezoneSafe(date, 'Europe/London', 'UTC')
+            expect(result).toEqual('2025-01-01')
+        })
+        it('should handle the user sending a nonsense timezone', () => {
+            const date = new Date('2025-01-01T10:00:00Z').getTime()
+            const result = toYYYYMMDDInTimezoneSafe(date, 'Not/A/Timezone', 'UTC')
+            expect(result).toEqual('2025-01-01')
+        })
+    })
```

Comment:
Can you step this moves across days as expected, by using -14 and +12 timezones? I know this is basically testing the library implementation, but it's important that we detect cross-date boundaries for this specific code

Reply:
Ah, I see you're testing `eventDifferentDay` below, so this might not be needed

Reply:
Done!

Reply:
I added some more time zones to these tests!